modelon/impact/client/client.py

```diff
@@ -149,9 +149,11 @@ def __init__(
         self._uri = URI(url)
         self._sal = modelon.impact.client.sal.service.Service(self._uri, context)
 
-        try:
-            self._validate_compatible_api_version()
-        except modelon.impact.client.sal.exceptions.AccessingJupyterHubError:
+        if self._sal.is_jupyterhub_url():
```

Comment:
One issue with doing this call before the `_validate_compatible_api_version` call is that you do not get the nicer error handling that re-throws the `except modelon.impact.client.sal.exceptions.CommunicationError`. There should probably be some error handling in `is_jupyterhub_url`, so that it always return a boolean and not exceptions.